File: frontend/src/router/modulos.ts
import type { ComponentType } from 'react'
import { Inicio } from '../modules/inicio/Inicio'
import { ProductosListado } from '../modules/productos/ProductosListado'
import { Combos } from '../modules/productos/Combos'
import { Aumentos } from '../modules/productos/Aumentos'
import { Historial } from '../modules/productos/Historial'
import { Stock } from '../modules/inventario/Stock'
import { RankingRentabilidad } from '../modules/inventario/RankingRentabilidad'
import { PosPage } from '../modules/pos/PosPage'
import { ControlCaja } from '../modules/caja/ControlCaja'
import { Movimientos } from '../modules/caja/Movimientos'
import { CuentasYCajas } from '../modules/caja/CuentasYCajas'
import { HistorialSesiones } from '../modules/caja/HistorialSesiones'
import { HistorialVentas } from '../modules/ventas/HistorialVentas'
import { Tickets } from '../modules/ventas/Tickets'
import { Panel } from '../modules/estadisticas/Panel'
import { Rankings } from '../modules/estadisticas/Rankings'
import { Rentabilidad } from '../modules/estadisticas/Rentabilidad'
import { VerdadDelNegocio } from '../modules/estadisticas/VerdadDelNegocio'
import { Resultado } from '../modules/contabilidad/Resultado'
import { MesAMes } from '../modules/contabilidad/MesAMes'
import { DeudasAging } from '../modules/contabilidad/DeudasAging'
import { Proveedores } from '../modules/proveedores/Proveedores'
import { Compras } from '../modules/compras/Compras'
import { PedidosProveedorTab } from '../modules/compras/PedidosProveedorTab'
import { GastosFijos, GastosVariables } from '../modules/finanzas/Gastos'
import { Repartos } from '../modules/repartos/Repartos'
import { Presupuestos } from '../modules/presupuestos/Presupuestos'
import { Asistente } from '../modules/asistente/Asistente'
import { Clientes } from '../modules/clientes/Clientes'
import { Sucursales } from '../modules/admin/Sucursales'
import { Turnos } from '../modules/empleados/Turnos'
import { Etiquetas } from '../modules/etiquetas/Etiquetas'
import { Config } from '../modules/config/Config'

/** Qué componente rinde cada ruta. Vive acá y no en App.tsx porque también lo
 * necesita SeccionPage para resolver la pestaña activa de una sección; si
 * siguiera en App.tsx el import sería circular.
 *
 * Lo que no figure acá cae en <ModulePlaceholder> (Mapa Neural, Kubobots…). */
export const MODULOS_IMPLEMENTADOS: Record<string, ComponentType> = {
  '/home': Inicio,
  '/pos': PosPage,
  '/asistente': Asistente,
  '/productos/listado': ProductosListado,
  '/productos/combos': Combos,
  '/productos/aumentos': Aumentos,
  '/productos/historial': Historial,
  '/stock': Stock,
  '/inventario/ranking': RankingRentabilidad,
  '/caja/contenedores': ControlCaja,
  '/caja/movimientos': Movimientos,
  '/caja/cuentas-cajas': CuentasYCajas,
  '/caja/historial': HistorialSesiones,
  '/ventas/historial': HistorialVentas,
  '/ventas/tickets': Tickets,
  '/estadisticas/panel': Panel,
  '/estadisticas/rankings': Rankings,
  '/estadisticas/rentabilidad': Rentabilidad,
  '/verdad-del-negocio': VerdadDelNegocio,
  '/contabilidad/resultado': Resultado,
  '/contabilidad/mes-a-mes': MesAMes,
  '/contabilidad/deudas': DeudasAging,
  '/proveedores': Proveedores,
  '/compras/registro': Compras,
  '/compras/gastos-variables': GastosVariables,
  '/compras/gastos-fijos': GastosFijos,
  '/compras/pedidos': PedidosProveedorTab,
  '/repartos': Repartos,
  '/presupuestos': Presupuestos,
  '/clientes': Clientes,
  '/sucursales': Sucursales,
  '/empleados': Turnos,
  '/etiquetas': Etiquetas,
  '/config': Config,
}

File: backend/core/modulos.py
"""Catálogo de módulos que el Dueño puede habilitar o deshabilitar por empleado.

`clave` es la ruta del frontend (router/navigation.ts), así que la misma lista
sirve para filtrar el menú y para cortar las requests en el servidor: no hay dos
listas que se puedan desincronizar.

`prefijos` son las rutas de la API que pertenecen SÓLO a ese módulo. Varios
módulos comparten endpoints con el POS —vender necesita leer productos, clientes
y cuentas de pago—, así que esos endpoints no se listan en ningún lado: apagar
"Productos" saca la pantalla de productos del menú, pero el buscador del POS
sigue funcionando. Lo que se corta en el servidor es lo exclusivo del módulo,
que es justo lo sensible (contabilidad, estadísticas, compras, la verdad del
negocio).
"""

# Inicio y Config no entran: sin Inicio el empleado entra a una pantalla vacía,
# y en Config lo único que ve un empleado es su propia cuenta y contraseña.
MODULOS = [
    {"clave": "/pos", "label": "Venta (POS)", "prefijos": []},
    {"clave": "/ventas", "label": "Historial de ventas", "prefijos": []},
    {"clave": "/presupuestos", "label": "Presupuestos", "prefijos": ["/api/presupuestos/"]},
    {"clave": "/repartos", "label": "Repartos", "prefijos": ["/api/repartos/"]},
    {"clave": "/clientes", "label": "Clientes", "prefijos": ["/api/crm/"]},
    {"clave": "/productos", "label": "Productos", "prefijos": [
        "/api/combos/", "/api/listas-precios/", "/api/ajustes-precios/",
        "/api/categorias-productos/", "/api/subcategorias-productos/",
        "/api/productos-universal/",
    ]},
    {"clave": "/stock", "label": "Stock", "prefijos": [
        "/api/inventario/depositos/", "/api/inventario/stock-deposito/",
        "/api/inventario/resumen/",
    ]},
    {"clave": "/inventario/ranking", "label": "Ranking de rentabilidad",
     "prefijos": ["/api/inventario/ranking-rentabilidad/"]},
    {"clave": "/etiquetas", "label": "Etiquetas", "prefijos": []},
    {"clave": "/proveedores", "label": "Proveedores", "prefijos": ["/api/proveedores/"]},
    {"clave": "/compras", "label": "Compras y gastos", "prefijos": [
        "/api/compras/", "/api/gastos/", "/api/pedidos/",
    ]},
    {"clave": "/caja", "label": "Caja", "prefijos": ["/api/caja/movimientos/"]},
    {"clave": "/estadisticas", "label": "Estadísticas", "prefijos": [
        "/api/estadisticas/panel/", "/api/estadisticas/rankings/",
        "/api/estadisticas/rentabilidad/",
    ]},
    {"clave": "/contabilidad", "label": "Contabilidad", "prefijos": ["/api/estadisticas/contabilidad/"]},
    {"clave": "/verdad-del-negocio", "label": "Verdad del Negocio",
     "prefijos": ["/api/estadisticas/verdad-del-negocio/"]},
    {"clave": "/empleados", "label": "Empleados y turnos", "prefijos": ["/api/empleados-turnos/"]},
    # Sin prefijos propios: la API de sucursales ya es sólo para el Dueño
    # (IsDueño), acá sólo se saca del menú.
    {"clave": "/sucursales", "label": "Sucursales", "prefijos": []},
    {"clave": "/asistente", "label": "Asistente", "prefijos": ["/api/asistente/"]},
]

CLAVES = {m["clave"] for m in MODULOS}


def modulo_de_ruta(path):
    """Clave del módulo dueño de esta ruta de API, o None si no es exclusiva
    de ninguno (endpoints compartidos con el POS, /auth/, /admin/)."""
    for modulo in MODULOS:
        for prefijo in modulo["prefijos"]:
            if path.startswith(prefijo):
                return modulo["clave"]
    return None
